lib/Mojo/Message/Request.pm

```diff
@@ -6,13 +6,14 @@ use Mojo::Cookie::Request;
 use Mojo::Util qw(b64_encode b64_decode sha1_sum);
 use Mojo::URL;
 
-my ($SEED, $COUNTER) = ($$ . time . rand, int rand 0xffffff);
-
 has env    => sub { {} };
 has method => 'GET';
 has [qw(proxy reverse_proxy)];
 has request_id => sub {
-  my $b64 = substr(sha1_base64($SEED . ($COUNTER = ($COUNTER + 1) % 0xffffff)), 0, 8);
+  state $seed    = $$ . time . rand;
+  state $counter = int rand 0xffffff;
```

Comment:
Can this be one line?

Reply:
Sadly no, state declarations don't support list assignment.

Reply:
Although I guess it could be written `state $seed = ..., state $counter = ...;` but that's kind of weird...

Reply:
Yeah, I tried to do `state ($seed, $counter) = ...` — and even `(state $seed, state $counter) = ...` — but no dice.  Quoth `perldiag`, it's a big ol' (F)atal error:

```
    Initialization of state variables in list currently forbidden
        (F) "state" only permits initializing a single variable, specified
        without parentheses. So "state $a = 42" and "state @a = qw(a b c)"
        are allowed, but not "state ($a) = 42" or "(state $a) = 42". To
        initialize more than one "state" variable, initialize them one at a
        time.
```

And I agree that `state $seed = ..., state $counter = ...` is awkward; you might as well just use a semicolon at that point, hehe!